crates/wdk-build/src/cargo_make.rs

```diff
@@ -754,16 +840,106 @@ fn load_wdk_build_makefile<S: AsRef<str> + AsRef<Utf8Path> + AsRef<Path>>(
     Ok(())
 }
 
+/// Get [`cargo_metadata::Metadata`] based off of manifest in
+/// `CARGO_MAKE_WORKING_DIRECTORY`
+///
+/// # Errors
+///
+/// This function will return a [`cargo_metadata::Error`] if `cargo_metadata`
+/// fails
+///
+/// # Panics
+///
+/// This function will panic if executed outside of a `cargo-make` task
+pub fn get_cargo_metadata() -> cargo_metadata::Result<Metadata> {
+    let manifest_path = {
+        let mut p: PathBuf = std::path::PathBuf::from(
+            std::env::var("CARGO_MAKE_WORKING_DIRECTORY")
+                .expect("CARGO_MAKE_WORKING_DIRECTORY should be set by cargo-make"),
+        );
+        p.push("Cargo.toml");
+        p
+    };
+
+    cargo_metadata::MetadataCommand::new()
+        .manifest_path(manifest_path)
+        .exec()
+}
+
+/// Execute a `FnOnce` closure, and handle its contents in a way compatible with
+/// `cargo-make`'s `condition_script`:
+/// 1. If the closure panics, the panic is caught and it returns an `Ok(())`.
+///    This ensures that panics encountered in `condition_script_closure` will
+///    not default to skipping the task.
+/// 2. If the closure executes without panicking, forward the result to
+///    `cargo-make`. `Ok` types will result in the task being run, and `Err`
+///    types will print the `Err` contents and then skip the task.
+///
+/// If you want your task to be skipped, return an `Err` from
+/// `condition_script_closure`. If you want the task to execute, return an
+/// `Ok(())` from `condition_script_closure`
+///
+/// # Errors
+///
+/// This function returns an error whenever `condition_script_closure` returns
+/// an error
+///
+/// # Panics
+///
+/// Panics if `CARGO_MAKE_CURRENT_TASK_NAME` is not set in the environment
+pub fn condition_script<F, E>(condition_script_closure: F) -> anyhow::Result<(), E>
+where
+    F: FnOnce() -> anyhow::Result<(), E> + UnwindSafe,
+{
+    std::panic::catch_unwind(condition_script_closure).unwrap_or_else(|_| {
+        // Note: Any panic messages has already been printed by this point
+
+        let cargo_make_task_name = env::var(CARGO_MAKE_CURRENT_TASK_NAME_ENV_VAR)
+            .expect("CARGO_MAKE_CURRENT_TASK_NAME should be set by cargo-make");
+
+        eprintln!(
+            "`condition_script` for {cargo_make_task_name} task panicked while executing. \
+             Defaulting to running {cargo_make_task_name} task."
+        );
+        Ok(())
+    })
+}
+
+/// `cargo-make` condition script for `package-driver-flow` task in
+/// [`rust-driver-makefile.toml`](../rust-driver-makefile.toml)
+///
+/// # Errors
+///
+/// This function returns an error whenever it determins that the
+/// `package-driver-flow` `cargo-make` task should be skipped (i.e. when no WDK
+/// configurations are detected)
+pub fn package_driver_flow_condition_script() -> anyhow::Result<()> {
+    condition_script(|| {
+        match metadata::Wdk::try_from(&get_cargo_metadata()?) {
```

Comment:
This doesn't work for workspaces that have WDK configs, but when the current crate being processed is not marked as a driver

Reply:
fixed